app/components/post_body_additional_content/index.js

```diff
@@ -14,18 +14,21 @@ import {getBool} from 'mattermost-redux/selectors/entities/preferences';
 import {ViewTypes} from 'app/constants';
 import {getDimensions} from 'app/selectors/device';
 import {getTheme} from 'app/selectors/preferences';
+import {extractFirstLink} from 'app/utils/url';
 
 import PostBodyAdditionalContent from './post_body_additional_content';
 
 function mapStateToProps(state, ownProps) {
     const config = getConfig(state);
     const previewsEnabled = getBool(state, Preferences.CATEGORY_ADVANCED_SETTINGS, `${ViewTypes.FEATURE_TOGGLE_PREFIX}${ViewTypes.EMBED_PREVIEW}`);
+    const link = extractFirstLink(ownProps.message);
```

Comment:
I think this is getting repeatedly called whenever the store changes which has me a bit worried. Then again, it's a pretty cheap calculation, so I don't think it's We could do something like
```javascript
function makeGetFirstLink() {
    let link;
    let lastMessage;

    return (message) => {
        if (message !== lastMessage) {
            link = extractFirstLink(message);
            lastMessage = message;
        }

        return link;
    };
}

function makeMapStateToProps() {
    const getFirstLink = makeGetFirstLink();
    
    function mapStateToProps(state, ownProps) {
        const link = getFirstLink(ownProps.message);

        return ...;
    }
}
```

Reply:
what if the post is edited and the link changes?